contract/src/main/kotlin/nl/nuts/consent/state/BranchState.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ *     Nuts consent cordapp
+ *     Copyright (C) 2020 Nuts community
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package nl.nuts.consent.state
+
+import net.corda.core.serialization.CordaSerializable
+
+/**
+ * BranchState indicates the state a ConsentBranch is in. Designed to be used for closing and errors.
+ */
+@CordaSerializable
+enum class BranchState {
+    Open,
+    Error,
+    Closed,
+    ClosedByPeer,
```

Comment:
Why is closed by peer a different state? Who is this Peer? Is it the party which is not the initiator? Should `closedBy` be a seperate property instead of something that must be derived from the state and the initiator field?

Reply:
and if there are more than 2 parties, that would then also not work....

Then maybe just closed and define thath the closingReason should contain the legalEntity and node uri of the party that closed it.